spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -918,15 +921,14 @@ def handle_go_to_definition(self, position):
 
     # ------------- LSP: Save/close file -----------------------------------
 
-    @request(method=LSPRequestTypes.DOCUMENT_WILL_SAVE,
+    @request(method=LSPRequestTypes.DOCUMENT_DID_SAVE,
              requires_response=False)
     def notify_save(self):
-        if self.will_save_notify:
-            params = {
-                'file': self.filename,
-                'reason': TextDocumentSaveReason.MANUAL
-            }
-            return params
+        # self.document_did_change()
+        params = {'file': self.filename}
+        if self.save_include_text:
+            params['text'] = self.toPlainText()
+        return params

```

Comment:
Why this change is needed? It's very different from the original one.

Reply:
The format of the LSP `textDocument/didSave` message is slightly different than `textDocument/willSave`.  This function was originally written to send the `willSave` message, but that's not the message I think we intended to send.  The `didSave` message also includes the `text` parameter optionally depending on the capabilities of the server, hence the conditional statement.

Reply:
Thanks for the explanation. It makes sense.